src/achievements.cpp

```diff
@@ -89,20 +84,27 @@ config achievements::read_achievements_file(const std::string& path)
  */
 void achievements::process_achievements_file(const config& cfg, const std::string& content_source)
 {
-	for(const config& c : cfg.child_range("achievement")) {
-		const std::string content_for = c["content_for"].str();
-		const std::string id = c["id"].str();
+	for(const config& achgrp : cfg.child_range("achievement_group")) {
```

Comment:
So this implementation implies:

* `[achievement_group]`s are not nestable
* There is no "implicit" `[achievement_group]` for an add-on. That is, an add-on with only a few achievements can't simply omit the tag and just drop `[achievement]` tags at toplevel.

I'm mentioning this mainly because it differs from the vision I had. Whether these are actually an issue, I'm not sure.

Reply:
> [achievement_group]s are not nestable

I'm not really clear what the benefit of allowing this would be.

> There is no "implicit" [achievement_group] for an add-on. That is, an add-on with only a few achievements can't simply omit the tag and just drop [achievement] tags at toplevel.

Needing to add an outer [achievement_group] even if there's only a few achievements doesn't seem like an issue to me.

Reply:
> I'm not really clear what the benefit of allowing this would be.

I guess the benefit would be that the group IDs have a lower chance of conflicting across add-ons? That's not that huge of a benefit tho…

Reply:
My guess/assumption would be that if there's an unintended conflict (as opposed to an intentional one where one add-on is adding more achievements to other content), UMC authors would add a prefix or a suffix to their IDs rather than nesting their `[achievement_group]`s.